### ISSUE TITLE
Add accounts resolution for associated token accounts

### ISSUE THREAD

**Issue by metaproph3t** [coral-xyz/anchor#2925]
Associated token accounts have statically derivable addresses, so it would be cool if the accounts resolver could do this and I didn't need to pass them into `.accounts({ ... })`. I'm guessing this would require adding the same `pda` field to the associated token accounts in the IDL.

**Comment by acheroncrypto** [comment]
Yeah, this would be great to have and is already on my todo list.
